### PR TITLE
Remotecfg reserved system attributes

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/reference/config-blocks/remotecfg.md [hunk 1]
```diff
@@ -50,6 +50,12 @@ If not set, the self-reported `id` that {{< param "PRODUCT_NAME" >}} uses is a r
 The `id` and `attributes` fields are used in the periodic request sent to the
 remote endpoint so that the API can decide what configuration to serve.
 
+The `attribute` map keys can include any custom value with the exception of a
+reserved prefix `collector_system_attribute_`. This reserved label prefix is
+used for automatic system attributes which cannot be overridden.
+* `collector_system_attribute_os`: The operating system where alloy is running.
+* `collector_system_attribute_version`: The version of alloy.
```
Comment (clayton-cornell) [inline docs/sources/reference/config-blocks/remotecfg.md:57]:
```suggestion
The `attribute` map keys can include any custom value with the exception of the reserved prefix `collector_system_attribute_`.
This reserved label prefix is used for automatic system attributes which cannot be overridden.

* `collector_system_attribute_os`: The operating system where {{< param "PRODUCT_NAME" >}} is running.
* `collector_system_attribute_version`: The version of {{< param "PRODUCT_NAME" >}}.
```

File: docs/sources/reference/config-blocks/remotecfg.md [hunk 2]
```diff
@@ -50,6 +50,12 @@ If not set, the self-reported `id` that {{< param "PRODUCT_NAME" >}} uses is a r
 The `id` and `attributes` fields are used in the periodic request sent to the
 remote endpoint so that the API can decide what configuration to serve.
 
+The `attribute` map keys can include any custom value with the exception of the reserved prefix `collector_system_attribute_`.
+This reserved label prefix is used for automatic system attributes which cannot be overridden.
```
Comment (clayton-cornell) [inline docs/sources/reference/config-blocks/remotecfg.md:54]:
```suggestion
This reserved label prefix is used for automatic system attributes which can't be overridden.
```

**Review by clayton-cornell — APPROVED** [review]

File: docs/sources/reference/config-blocks/remotecfg.md (hunk 2 above)
Comment (clayton-cornell) [reply docs/sources/reference/config-blocks/remotecfg.md:54]: Approving for docs

File: docs/sources/reference/config-blocks/remotecfg.md [hunk 3]
```diff
@@ -50,6 +50,12 @@ If not set, the self-reported `id` that {{< param "PRODUCT_NAME" >}} uses is a r
 The `id` and `attributes` fields are used in the periodic request sent to the
 remote endpoint so that the API can decide what configuration to serve.
 
+The `attribute` map keys can include any custom value with the exception of the reserved prefix `collector.`.
```
Comment (clayton-cornell) [inline docs/sources/reference/config-blocks/remotecfg.md:53]:
```suggestion
The `attribute` map keys can include any custom value except the reserved prefix `collector.`.
```

File: docs/sources/reference/config-blocks/remotecfg.md [hunk 4]
```diff
@@ -50,6 +50,12 @@ If not set, the self-reported `id` that {{< param "PRODUCT_NAME" >}} uses is a r
...
+This reserved label prefix is used for automatic system attributes which can't be overridden.
```
Comment (clayton-cornell) [inline docs/sources/reference/config-blocks/remotecfg.md:54]:
```suggestion
The reserved label prefix is for automatic system attributes.
You can't override this prefix.
```

**Review by tpaschalis — APPROVED** [review]
LGTM!

**Pull request merged**